src/digraph-cyclicity-hadwick-james.jl

```diff
@@ -0,0 +1,99 @@
+@doc_str """
+Find circuits (including self-loops) using the algorithm of Hadwick & James
+("Enumerating Circuits and Loops in Graphs with Self-Arcs and Multiple-Arcs", 2008)
+"""
+function simplecycles_hadwick_james end
+@traitfn function simplecycles_hadwick_james(g::::IsDirected)
+    nvg = nv(g)
+    T = eltype(g)
+    B = Vector{Vector{T}}()
+    for i in vertices(g)
+        push!(B, Vector{T}())
+    end
+    blocked = falses(nvg)
+    stack = T[]
+    cycles = Vector{Vector{T}}()
+    for v in vertices(g)
+        circuit_recursive!(g, v, v, blocked, B, stack, cycles)
+        resetblocked!(blocked)
+        resetB!(B)
+    end
+    return cycles
+end
+
+"""
+Reset B work structure.
+"""
+resetB!(B) = map!(empty!, B, B)
+
+"""
+Reset vector of "blocked" vertices.
+"""
+resetblocked!(blocked) = fill!(blocked, false)
+
+@doc_str """
+Find circuits starting from v1 recursively.
+"""
```

Comment:
I see. I've never had to use the `@doc` string macro; thought it had to be included because of SimpleTraits. Thanks for the clarification!